### ISSUE TITLE
Better support for File arguments

### ISSUE THREAD

**Issue by Kiwi** [scallop/scallop#69]
At the moment you can use:

<code>implicit val fileConverter = singleArgConverter(fname => new File(fname)).flatMap(f => if (f.exists) Right(Some(f)) else Left(Unit))</code>

along with <code>val someFile = opt[File]()</code>

but this would produce unhelpful error messages. It would be nice to have something similar to what <a href="http://docs.python.org/dev/library/argparse.html#filetype-objects">argparse</a> in python does for you, which goes far as to specify file mode, among other things.

**Comment by Rogach** [comment]
I implemented a change, and published a snapshot (for scala 2.10.x):

``` scala
resolvers += "sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"
libraryDependencies += "org.rogach" %% "scallop" % "0.9.3~1~SNAPSHOT"
```

**Comment by Rogach** [comment]
I'm closing the issue and merging the change now. If you will have more comments, please let me know!
